core/src/main/java/io/confluent/rest/Application.java

```diff
@@ -121,10 +125,77 @@ protected void doStop() throws Exception {
       }
     };
 
-    NetworkTrafficServerConnector connector = new NetworkTrafficServerConnector(server);
-    connector.addNetworkTrafficListener(new MetricsListener(metrics, "jetty", metricTags));
-    connector.setPort(getConfiguration().getInt(RestConfig.PORT_CONFIG));
-    server.setConnectors(new Connector[]{connector});
+    // http
+    if (config.getString(RestConfig.REST_PROTOCOL_CONFIG).equals(RestConfig.REST_PROTOCOL_HTTP) ||
+        config.getString(RestConfig.REST_PROTOCOL_CONFIG).equals(RestConfig.REST_PROTOCOL_HTTP_PLUS_HTTPS)) {
+      NetworkTrafficServerConnector connector = new NetworkTrafficServerConnector(server);
+      connector.addNetworkTrafficListener(new MetricsListener(metrics, "jetty", metricTags));
+      connector.setPort(getConfiguration().getInt(RestConfig.PORT_CONFIG));
+      server.addConnector(connector);
+      log.info("Added http connector.");
+    }
+
+    // https
+    if (config.getString(RestConfig.REST_PROTOCOL_CONFIG).equals(RestConfig.REST_PROTOCOL_HTTPS) ||
+        config.getString(RestConfig.REST_PROTOCOL_CONFIG).equals(RestConfig.REST_PROTOCOL_HTTP_PLUS_HTTPS)) {
+      SslContextFactory sslContextFactory = new SslContextFactory();
+      // IMPORTANT: the key's CN, stored in the keystore, must match the FQDN. This is a Jetty requirement.
```

Comment:
Actually, I may have generated the cert incorrectly. There may be a nasty bug with `keytool` where certain command-line args aren't forwarded using `genkey` instead of `genkeypair`. I'm investigating.